Description: Create a Makefile
 This patch generate a Makefile without configure.curvedns.
 It also update cache_hashtable.h dnscurve.h and curvedns-keygen.c with sodium.h.
 This patch is based on Leo Vandewoestijne's work (vsevolod@FreeBSD.org).
Author: Stephane Neveu
Last-Update: 2017-06-22
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,67 @@
+# CurveDNS Debian Makefile (link against libsodium)
+
+EXTRALIB=-lev -lsodium
+
+TARGETS=curvedns-keygen curvedns
+
+.PHONY: targets clean distclean install
+
+targets: $(TARGETS)
+
+clean:
+	rm -f *.a *.o $(TARGETS)
+
+install:
+	@echo Sorry, no automated install. Copy the following binaries to your preferred destination path:
+	@echo "  $(TARGETS)"
+
+distclean: clean
+
+debug.o: debug.c debug.h
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c debug.c
+
+cache_hashtable.o: cache_hashtable.c cache_hashtable.h debug.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c cache_hashtable.c
+
+# ready for possible critbit addition
+cache.a: cache_hashtable.o
+	$(AR) cr cache.a cache_hashtable.o
+	ranlib cache.a
+
+dns.o: dns.c dns.h debug.o event.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c dns.c
+
+dnscurve.o: dnscurve.c dnscurve.h debug.o event.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c dnscurve.c
+
+curvedns.o: curvedns.c curvedns.h debug.o ip.o misc.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c curvedns.c
+
+ip.o: ip.c ip.h debug.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c ip.c
+
+event_tcp.o: event_tcp.c event.h debug.o ip.o cache.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c event_tcp.c
+
+event_udp.o: event_udp.c event.h debug.o ip.o cache.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c event_udp.c
+
+event_main.o: event_main.c event.h debug.o ip.o cache.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c event_main.c
+
+event.a: event_main.o event_udp.o event_tcp.o
+	$(AR) cr event.a event_main.o event_udp.o event_tcp.o
+	ranlib event.a
+
+misc.o: misc.c misc.h ip.o debug.o
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c misc.c
+
+curvedns-keygen.o: curvedns-keygen.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c curvedns-keygen.c
+
+# The targets:
+curvedns: debug.o ip.o misc.o cache.a event.a dnscurve.o dns.o curvedns.o
+	$(CC) $(LDFLAGS) debug.o ip.o misc.o dnscurve.o dns.o cache.a event.a curvedns.o $(EXTRALIB) -o curvedns
+
+curvedns-keygen: curvedns-keygen.o debug.o ip.o misc.o
+	$(CC) $(LDFLAGS) curvedns-keygen.o debug.o ip.o misc.o $(EXTRALIB) -o curvedns-keygen
--- a/cache_hashtable.h
+++ b/cache_hashtable.h
@@ -43,7 +43,7 @@
 #include <string.h>
 
 #include <ev.h>
-#include "crypto_box_curve25519xsalsa20poly1305.h"
+#include "sodium.h"
 #include "debug.h"
 
 // The cache mechanism isn't really general, it is focused
--- a/curvedns-keygen.c
+++ b/curvedns-keygen.c
@@ -41,7 +41,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
-#include "crypto_box_curve25519xsalsa20poly1305.h"
+#include "sodium.h"
 #include "debug.h"
 #include "misc.h"
 
--- a/dnscurve.h
+++ b/dnscurve.h
@@ -43,7 +43,7 @@
 #include <string.h>
 #include <errno.h>
 
-#include "crypto_box_curve25519xsalsa20poly1305.h"
+#include "sodium.h"
 #include "debug.h"
 #include "event.h"
 
